python/numpydoc.rst

```diff
@@ -589,9 +589,15 @@ This works for both built-in types and most importable objects:
        [...]
 
 In general, provide the full namespace to the object, such as ```lsst.pipe.base.Struct```.
-It may be possible to reference objects in the same namespace as the current module without any namespace prefix.
+It may be possible to reference objects in the same package as the current type or function without any namespace prefix.
 Always check the compiled documentation site to ensure the link worked.
 
+.. note::
+
+   Relative links to types are evaluated relative to the *package* in which the linking object is documented, not the module file in which the docstring is written.
```

Comment:
I also saw in lsst/daf_butler#846 that a `.` prefix can be used in numpydoc to allow relative to current.